ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneResults/CloneResults.tsx

```diff
@@ -64,6 +65,23 @@ const getErrorDescription = (code: ValueOf<typeof CloneErrorCodes>) => {
   }
 };
 
+const getSystemId = (error: CloneError["destinations"][0]) => {
+  if ("system_id" in error) {
+    return error.system_id as string;
+  } else if (error.code === CloneErrorCodes.ITEM_INVALID) {
+    // Item invalid errors short circuit the clone operation, and the API is
+    // unable to send through a system_id without some significant reworking of
+    // how the Django forms work. For this case we extract the system_id from
+    // the error message manually.
+    return (
+      error.message.match(
+        /Machine [0-9]+ is invalid: Select a valid choice\. (.{6}) is not one of the available choices\./
+      )?.[1] || ""
+    );
```

Comment:
This is obviously not ideal, but it's the only way to extract the `system_id` used for grouping/filtering for now.